tests/lac/tensor_product_matrix.h

```diff
@@ -0,0 +1,133 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#pragma once
+
+#include <deal.II/fe/fe_nothing.h>
+#include <deal.II/fe/fe_values.h>
+
+namespace dealii
+{
+  namespace GridTools
+  {
+    // Compute harminic extend of all non-artificial cells.
+    template <int dim>
+    std::vector<std::array<double, dim>>
+    compute_harmonic_cell_extend(const Mapping<dim> &       mapping,
```

Comment:
- some utility functions that I needed and plan to integrate into the library in future PRs.